### PR TITLE
Coverage for instance type resource using linter mnd

### LINKED ISSUE #12662: linter: Cover instancetypes source files using the 'mnd' linter
**Is your feature request related to a problem? Please describe**:

As part of golang-ci linters update in https://github.com/kubevirt/kubevirt/pull/12600, there was a need to exclude some instancetype source files for the 'mnd' linter.

**Describe the solution you'd like**:

Remove the instancetype source files from the 'mnd' ignored files by solving the errors they produce.

**Describe alternatives you've considered**:

**Additional context**:

### PR DESCRIPTION
**Pull request by anishbista60**
### What this PR does
Before this PR:
there was a need to exclude some instancetype source files for the 'mnd' linter.
After this PR:
Remove the instancetype source files from the 'mnd' ignored files by solving the errors they produce.

Fixes #12662 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by anishbista60** [comment]
@EdDev works fine in my machine . Could you please take a look ?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @anishbista60 
Please see the inline comments.

File: tests/instancetype/instancetype.go [hunk 1]
```diff
@@ -1706,9 +1716,9 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									CPU: &virtv1.CPU{
-										Cores:   uint32(2),
+										Cores:   uint32(MemorySize2),
 										Threads: uint32(1),
-										Sockets: uint32(2),
+										Sockets: uint32(MemorySize2),
```
[the remark this answers is not in the document]
Comment (anishbista60) [reply tests/instancetype/instancetype.go:1721]: @orelmisan should i revert the changes for use other name .

File: tests/instancetype/instancetype.go [hunk 2]
```diff
@@ -401,7 +411,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 
 			var apiStatus errors.APIStatus
 			Expect(goerrors.As(err, &apiStatus)).To(BeTrue(), "error should be type APIStatus")
-			Expect(apiStatus.Status().Details.Causes).To(HaveLen(3))
+			Expect(apiStatus.Status().Details.Causes).To(HaveLen(expectedCausesLength))
```
[the remark this answers is not in the document]
Comment (anishbista60) [reply tests/instancetype/instancetype.go:414]: @orelmisan do you mean like this . 

```
expectedcauses := []metav1.StatusCause{
				{
					Type:    "",
					Message: ",
					Field:   "",
				},
			}
			Expect(apiStatus.Status().Details.Causes).To(HaveLen(expectedCausesLength))
			Expect(apiStatus.Status().Details.Causes).To(Equal(expectedcauses))
```
if yes , what to put into type, message and field . Could you please help me here ?

File: tests/instancetype/instancetype.go [hunk 3]
```diff
@@ -47,6 +47,19 @@ import (
...
+		timeout                                = 300 * time.Second
+		dvSuccessTimeoutSeconds                = 180
+		guestMemorySize                        = 2
+		numCore                                = 2
+		numSocket                              = 2
+		numThread                              = 2
+		MemorySize4                            = 4
+		memorySize2GB                          = 2 * 1024 * 1024 * 1024
+		memorySize1GB                          = 1 * 1024 * 1024 * 1024
+	)
```
Comment (0xFelix) [inline tests/instancetype/instancetype.go:62]: Are all these constants really necessary? I would rather only add constants for the places that gomnd is complaining about.

Comment (anishbista60) [reply tests/instancetype/instancetype.go:62]: @0xFelix  ok three constant are added additionally , I'll remove that

File: tests/instancetype/instancetype.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (anishbista60) [reply tests/instancetype/instancetype.go:417]: @0xFelix here could you please review

[the remark this answers is not in the document]
Comment (anishbista60) [reply tests/instancetype/instancetype.go:417]: > @0xFelix here could you please review?

**Comment by anishbista60** [comment]
@0xFelix could you please take look ? Thank you

File: tests/instancetype/instancetype.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lyarwood) [reply tests/instancetype/instancetype.go:415]: That's already done below but could be refactored in a follow up, I wouldn't change it as part of this PR.

File: tests/instancetype/instancetype.go [hunk 4]
```diff
@@ -47,6 +47,17 @@ import (
...
+		preferredTerminationGracePeriodSeconds = 15
+		expectedCausesLength                   = 3
+		timeout                                = 300 * time.Second
+		dvSuccessTimeoutSeconds                = 180
+		guestMemorySize                        = 2
+		numCPU                                 = 2
+		MemorySize4                            = 4
+		memorySize2GB                          = 2 * 1024 * 1024 * 1024
+		memorySize1GB                          = 1 * 1024 * 1024 * 1024
+	)
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:60]: Can you please move most of these constants down into their relevant contexts?

File: pkg/instancetype/instancetype.go [hunk 5]
```diff
@@ -802,7 +803,7 @@ func (m *InstancetypeMethods) InferDefaultInstancetype(vm *virtv1.VirtualMachine
 		var ignoreableInferenceErr *IgnoreableInferenceError
 		if errors.As(err, &ignoreableInferenceErr) && ignoreFailure {
 			//nolint:gomnd
```
Comment (lyarwood) [inline pkg/instancetype/instancetype.go:805]: Can you also drop this?

File: pkg/instancetype/instancetype.go [hunk 6]
```diff
@@ -838,7 +839,7 @@ func (m *InstancetypeMethods) InferDefaultPreference(vm *virtv1.VirtualMachine)
 
 		if errors.As(err, &ignoreableInferenceErr) && ignoreFailure {
 			//nolint:gomnd
```
Comment (lyarwood) [inline pkg/instancetype/instancetype.go:841]: As above.

File: tests/instancetype/instancetype.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (anishbista60) [reply tests/instancetype/instancetype.go:415]: @lyarwood ok sir

File: tests/instancetype/instancetype.go [hunk 7]
```diff
@@ -1426,6 +1433,13 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			preferThreads = instancetypev1beta1.Threads
 			preferAny     = instancetypev1beta1.Any
 		)
+		const (
+			memorySize2GB   = 2 * 1024 * 1024 * 1024
+			memorySize1GB   = 1 * 1024 * 1024 * 1024
+			guestMemorySize = 2
+			numCPU          = 2
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1440]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 8]
```diff
@@ -1449,7 +1463,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 					},
 					Spec: instancetypev1beta1.VirtualMachineInstancetypeSpec{
 						CPU: instancetypev1beta1.CPUInstancetype{
-							Guest: uint32(2),
+							Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1466]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 9]
```diff
@@ -1460,7 +1474,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 					Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(2),
+								Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1477]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 10]
```diff
@@ -1537,7 +1551,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1554]: `requiredNumvCPUs`?

File: tests/instancetype/instancetype.go [hunk 11]
```diff
@@ -1557,7 +1571,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 									CPU: &virtv1.CPU{
 										Cores:   uint32(1),
 										Threads: uint32(1),
-										Sockets: uint32(2),
+										Sockets: uint32(numCPU),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1574]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 12]
```diff
@@ -1577,7 +1591,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 						},
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(2),
+								Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1594]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 13]
```diff
@@ -1595,7 +1609,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									CPU: &virtv1.CPU{
-										Cores:   uint32(2),
+										Cores:   uint32(numCPU),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1612]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 14]
```diff
@@ -1648,7 +1662,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 						},
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(2),
+								Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1665]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 15]
```diff
@@ -1667,7 +1681,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 								Domain: virtv1.DomainSpec{
 									CPU: &virtv1.CPU{
 										Cores:   uint32(1),
-										Threads: uint32(2),
+										Threads: uint32(numCPU),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1684]: `providedvCPUs`?

File: tests/instancetype/instancetype.go [hunk 16]
```diff
@@ -1688,7 +1702,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 						},
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(4),
+								Guest: uint32(MemorySize4),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1705]: `requiredvCPUs`?

File: tests/instancetype/instancetype.go [hunk 17]
```diff
@@ -1706,9 +1720,9 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									CPU: &virtv1.CPU{
-										Cores:   uint32(2),
+										Cores:   uint32(numCPU),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1723]: Can you create new constants for these as total vCPU count is sockets * cores * threads.

Something like `providedCores` and `providedSockets`?

File: tests/instancetype/instancetype.go [hunk 18]
```diff
@@ -1788,7 +1802,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 					Spec: instancetypev1beta1.VirtualMachinePreferenceSpec{
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(2),
+								Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1805]: `requiredvCPUs` ?

File: tests/instancetype/instancetype.go [hunk 19]
```diff
@@ -1867,7 +1881,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1884]: `requiredvCPUs`?

File: tests/instancetype/instancetype.go [hunk 20]
```diff
@@ -1908,7 +1922,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 						},
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(2),
+								Guest: uint32(guestMemorySize),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1925]: `requiredvCPUs`?

File: tests/instancetype/instancetype.go [hunk 21]
```diff
@@ -1949,7 +1963,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1966]: `requiredvCPUs`?

File: tests/instancetype/instancetype.go [hunk 22]
```diff
@@ -1990,7 +2004,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 						},
 						Requirements: &instancetypev1beta1.PreferenceRequirements{
 							CPU: &instancetypev1beta1.CPUPreferenceRequirement{
-								Guest: uint32(4),
+								Guest: uint32(MemorySize4),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:2007]: `requiredvCPUs`?

File: tests/instancetype/instancetype.go [hunk 23]
```diff
@@ -2008,7 +2022,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									CPU: &virtv1.CPU{
-										Cores:   uint32(2),
+										Cores:   uint32(numCPU),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:2025]: `providedvCPUs` ?

File: tests/instancetype/instancetype.go [hunk 24]
```diff
@@ -2046,7 +2060,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									Memory: &virtv1.Memory{
-										Guest: resource.NewQuantity(1*1024*1024*1024, resource.BinarySI),
+										Guest: resource.NewQuantity(memorySize1GB, resource.BinarySI),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:2063]: Can you just change this into a var?

File: tests/instancetype/instancetype.go [hunk 25]
```diff
@@ -1743,7 +1757,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									Memory: &virtv1.Memory{
-										Guest: resource.NewQuantity(2*1024*1024*1024, resource.BinarySI),
+										Guest: resource.NewQuantity(memorySize2GB, resource.BinarySI),
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1760]: Can you just change this into a var?

File: tests/instancetype/instancetype.go [hunk 26]
```diff
@@ -47,6 +47,9 @@ import (
 
 var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute] Instancetype and Preferences", decorators.SigCompute, func() {
 	var virtClient kubecli.KubevirtClient
+	const (
+		timeout = 300 * time.Second
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:51]: `timeoutSeconds`?

Also can be a single line above `var`?

File: tests/instancetype/instancetype.go [hunk 27]
```diff
@@ -1426,6 +1433,13 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			preferThreads = instancetypev1beta1.Threads
 			preferAny     = instancetypev1beta1.Any
 		)
+		const (
+			memorySize2GB   = 2 * 1024 * 1024 * 1024
+			memorySize1GB   = 1 * 1024 * 1024 * 1024
+			guestMemorySize = 2
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1439]: This doesn't need to be defined.

**Review by lyarwood — CHANGES_REQUESTED** [review]

**Comment by anishbista60** [comment]
@lyarwood Done sir. Change it based on your recommendation .

File: tests/instancetype/instancetype.go [hunk 28]
```diff
@@ -1743,7 +1760,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 							Spec: virtv1.VirtualMachineInstanceSpec{
 								Domain: virtv1.DomainSpec{
 									Memory: &virtv1.Memory{
-										Guest: resource.NewQuantity(2*1024*1024*1024, resource.BinarySI),
+										Guest: resource.NewQuantity(memorySize2GB, resource.BinarySI),
```
Comment (orelmisan) [inline tests/instancetype/instancetype.go:1763]: Please consider using https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#MustParse:
```suggestion
										Guest: resource.MustParse("2Gi"),
```
Same in other places defining memory quantity.

File: tests/instancetype/instancetype.go [hunk 29]
```diff
@@ -704,11 +710,12 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
-			instancetype *instancetypev1beta1.VirtualMachineInstancetype
-			preference   *instancetypev1beta1.VirtualMachinePreference
-			sourceDV     *cdiv1beta1.DataVolume
-			namespace    string
+			vm                      *virtv1.VirtualMachine
+			instancetype            *instancetypev1beta1.VirtualMachineInstancetype
+			preference              *instancetypev1beta1.VirtualMachinePreference
+			sourceDV                *cdiv1beta1.DataVolume
+			namespace               string
+			dvSuccessTimeoutSeconds = 180
```
Comment (orelmisan) [inline tests/instancetype/instancetype.go:718]: Could this be a const?

File: tests/instancetype/instancetype.go [hunk 30]
```diff
@@ -297,7 +303,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			preference.Spec.Firmware = &instancetypev1beta1.FirmwarePreferences{
 				PreferredUseBios: pointer.P(true),
 			}
-			preference.Spec.PreferredTerminationGracePeriodSeconds = pointer.P(int64(15))
+			preference.Spec.PreferredTerminationGracePeriodSeconds = pointer.P(int64(preferredTerminationGracePeriodSeconds))
```
Comment (orelmisan) [inline tests/instancetype/instancetype.go:306]: KubeVirt has a generic pointer package:
https://github.com/kubevirt/kubevirt/blob/43531861ac5fc7795ac6523c482b2e2de74dfa3b/pkg/pointer/pointer.go#L22

Please consider using it.

**Review by orelmisan — COMMENTED** [review]
Thank you for the changes @anishbista60.

**Comment by anishbista60** [comment]
@orelmisan  please let me know if there is anything to look into.

**Comment by orelmisan** [comment]
> @orelmisan please let me know if there is anything to look into.

Thank you for the changes @anishbista60.
Could you please address https://github.com/kubevirt/kubevirt/pull/12694/files#r1741853940 ?
It was resolved without providing feedback for the concern that was raised.

**Comment by anishbista60** [comment]
> > @orelmisan please let me know if there is anything to look into.
> 
> Thank you for the changes @anishbista60. Could you please address https://github.com/kubevirt/kubevirt/pull/12694/files#r1741853940 ? It was resolved without providing feedback for the concern that was raised.

Regarding this, you told me to change `var` into `const`. But , when I raised this PR , I added  them as constant but later, on the recommendation of @lyarwood  I changed it to. `var`. Now, should I leave it as it is or change it. 

Thanykou

**Comment by orelmisan** [comment]
> > > @orelmisan please let me know if there is anything to look into.
> > 
> > 
> > Thank you for the changes @anishbista60. Could you please address https://github.com/kubevirt/kubevirt/pull/12694/files#r1741853940 ? It was resolved without providing feedback for the concern that was raised.
> 
> Regarding this, you told me to change `var` into `const`. But , when I raised this PR , I added them as constant but later, on the recommendation of @lyarwood I changed it to. `var`. Now, should I leave it as it is or change it ?
> 
> Thank you

@lyarwood is the owner of this file, what he says has precedence.

File: tests/instancetype/instancetype.go [hunk 31]
```diff
@@ -1422,9 +1429,19 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
 	Context("VirtualMachine using preference resource requirements", func() {
 		var (
-			preferCores   = instancetypev1beta1.Cores
-			preferThreads = instancetypev1beta1.Threads
-			preferAny     = instancetypev1beta1.Any
+			preferCores    = instancetypev1beta1.Cores
+			preferThreads  = instancetypev1beta1.Threads
+			preferAny      = instancetypev1beta1.Any
+			guestMemory2GB = resource.MustParse("2GiB")
+			guestMemory1GB = resource.MustParse("1GiB")
```
Comment (orelmisan) [inline tests/instancetype/instancetype.go:1436]: @anishbista60 the tests fail due to a panic here:
```suggestion
			guestMemory2GB = resource.MustParse("2Gi")
			guestMemory1GB = resource.MustParse("1Gi")
```

File: tests/instancetype/instancetype.go (hunk 26 above)
Comment (lyarwood) [reply tests/instancetype/instancetype.go:51]: This should just be:

```
const timeoutSeconds =  300 * time.Second
```

File: tests/instancetype/instancetype.go [hunk 32]
```diff
@@ -1422,9 +1429,19 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
+			preferCores    = instancetypev1beta1.Cores
+			preferThreads  = instancetypev1beta1.Threads
+			preferAny      = instancetypev1beta1.Any
+			guestMemory2GB = resource.MustParse("2Gi")
+			guestMemory1GB = resource.MustParse("1Gi")
+		)
+		const (
+			providedvCPUs    = 2
+			requiredNumvCPUs = 2
+			requiredvCPUs    = 4
```
Comment (lyarwood) [inline tests/instancetype/instancetype.go:1441]: I still think the naming could be better here...

**Comment by anishbista60** [comment]
@lyarwood Done . 
this name is there. Based on your suggestion in previous review 

```
providedvCPUs    = 2
requiredNumvCPUs = 2
requiredvCPUs    = 4
```

**Comment by lyarwood** [comment]
> @lyarwood Done . These name are present based on your suggestion in previous review
> 
> ```
> providedvCPUs    = 2
> requiredNumvCPUs = 2
> requiredvCPUs    = 4
> ```

Yup okay, for now this is good enough. Thanks for your contribution!

/approve

**Comment by lyarwood** [comment]
> @lyarwood Done . These name are present based on your suggestion in previous review
> 
> ```
> providedvCPUs    = 2
> requiredNumvCPUs = 2
> requiredvCPUs    = 4
> ```

Yup okay, for now this is good enough. Thanks for your contribution!

/lgtm

(previously tried to /approve but missed this touched `.golangci.yml`)

**Comment by anishbista60** [comment]
@lyarwood linter test case throws an error 

```
tests/instancetype/instancetype.go:50:8: ST1011: var timeoutSeconds is of type time.Duration; don't use unit-specific suffix "Seconds" (stylecheck)
	const timeoutSeconds = 300 * time.Second
```
so , i change it to `timeout`

**Comment by lyarwood** [comment]
/lgtm

TIL that's even a rule, odd that it's specific to `time.Duration` variables but I guess that makes sense.

**Comment by anishbista60** [comment]
@lyarwood  what is the reason of failing behind `pull-kubevirt-e2e-k8s-1.31-sig-compute` ?.

**Comment by 0xFelix** [comment]
It's a flaky test not related to your changes.

/retest-required

**Comment by anishbista60** [comment]
@lyarwood Could you please move further ?

**Comment by anishbista60** [comment]
> It's a flaky test not related to your changes.
> 
> /retest-required

Thankyou @0xFelix . I always remember your help at my First PR 7 months back. ❤️

**Comment by anishbista60** [comment]
anyone please look at this PR and try to get merge. Thannkyou

**Review by 0xFelix — COMMENTED** [review]
/test all

**Comment by anishbista60** [comment]
@lyarwood could you please move further ?

**Comment by anishbista60** [comment]
Please try to get it merge. only approval is needed. Thank you.

**Review by 0xFelix — COMMENTED** [review]
/approve
/test all

**Review by 0xFelix — APPROVED** [review]
Not able to approve because `.golangi.yml` was changed.

/lgtm

**Comment by anishbista60** [comment]
@jean-edouard Please have a look .

**Comment by anishbista60** [comment]
Thank you everyone who helped in reviewing

**Pull request merged**